src/OrleansRuntime/Silo/SiloHost.cs

```diff
@@ -249,6 +250,39 @@ public void ShutdownOrleansSilo()
         }
 
         /// <summary>
+        /// Returns a task that will resolve when the silo has finished shutting down, or the cancellation token is cancelled.
+        /// </summary>
+        /// <param name="millisecondsTimeout">Timeout, or -1 for infinite.</param>
+        /// <param name="cancellationToken">Token that cancels waiting for shutdown.</param>
+        /// <returns></returns>
+        public Task ShutdownOrleansSiloAsync(int millisecondsTimeout, CancellationToken cancellationToken)
+        {
+            if (orleans == null || !IsStarted)
+                return Task.CompletedTask;
+
+            IsStarted = false;
+
+            var shutdownThread = new Thread(o =>
```

Comment:
Why not Task.Run() instead of creating a thread? Not a big deal, just curious.

Reply:
Only to follow the style of the other shutdown code. We can change it if you want.